Wikipedia/Code/MWKSearchResult.m

```diff
@@ -235,4 +235,33 @@ - (nullable NSURL *)articleURLForSiteURL:(NSURL *)siteURL {
     return [siteURL wmf_URLWithTitle:self.displayTitle];
 }
 
+#pragma mark - NSObject
+
+- (BOOL)isEqualToSearchResult:(MWKSearchResult *)searchResult {
+    if (!searchResult) {
+        return NO;
+    }
+    
+    BOOL haveEqualDisplayTitles = (!self.displayTitle && !searchResult.displayTitle) || [self.displayTitle isEqualToString:searchResult.displayTitle];
+    
+    return haveEqualDisplayTitles;
+}
+
+- (BOOL)isEqual:(id)object {
+    if (self == object) {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:[MWKSearchResult class]]) {
+        return NO;
+    }
+    
+    return [self isEqualToSearchResult:(MWKSearchResult *)object];
+    
+}
+
+- (NSUInteger)hash {
+    return [self.displayTitle hash];
+}
```

Comment:
This solution works, but I'm not sure if overriding `isEqual` has other consequences for subclasses of `MTLModel`. The reason they aren't equal is due to the `index` property - the results have different index values so `MTLModel`'s `isEqual` implementation returns false. A lower impact fix would be to alter [mergeResultsFromModel:](https://github.com/wikimedia/wikipedia-ios/blob/dff6ea281f0fa262a008e49843680778215d5c57/Wikipedia/Code/WMFSearchResults.m#L98) to check for conflicting display titles rather than checking for `containsObject:`.